File: sdk/src/types/agent.ts
import { PublicKey } from '@solana/web3.js';

export interface AgentConfig {
  name: string;
  uri: string;
  strategyHash: Uint8Array;
  owner: PublicKey;
}

export interface AgentProfile {
  owner: PublicKey;
  mint: PublicKey;
  strategyHash: Uint8Array;
  eloRating: number;
  wins: number;
  losses: number;
  totalPnl: number;
  totalTrades: number;
  competitionsEntered: number;
  status: 'Active' | 'Suspended' | 'Retired';
  createdAt: number;
}

File: programs/arena/src/state/agent.rs
use anchor_lang::prelude::*;

#[derive(AnchorSerialize, AnchorDeserialize, Clone, Copy, PartialEq, Eq, InitSpace)]
pub enum AgentStatus {
  Active,
  Suspended,
  Retired,
}

#[account]
#[derive(InitSpace)]
pub struct Agent {
  pub owner: Pubkey,
  pub mint: Pubkey,
  pub strategy_hash: [u8; 32],
  pub elo_rating: u32,
  pub wins: u32,
  pub losses: u32,
  pub total_pnl: i64,
  pub total_trades: u32,
  pub competitions_entered: u32,
  pub status: AgentStatus,
  pub created_at: i64,
  pub bump: u8,
}
